pkg/syncer/cnsoperator/controller/cnsnodevmbatchattachment/cnsnodevmbatchattachment_helper.go

```diff
@@ -124,23 +134,43 @@ func getVolumesToDetachFromInstance(ctx context.Context,
 // Volumes in status: pvc-1
 // Volumes on vCenter: pvc-1
 func removeStaleEntriesFromInstanceStatus(ctx context.Context,
+	client client.Client,
 	instance *v1alpha1.CnsNodeVmBatchAttachment,
-	pvcsToDetach map[string]string, volumeNamesInSpec map[string]string) {
+	pvcsToDetach map[string]string, volumeNamesInSpec map[string]string) error {
 	log := logger.GetLogger(ctx)
 
+	// Initializes kubernetes client.
```

Comment:
Can we pass k8sClient in the arg and create only once instance of k8sClient in the reconcile.